### PR TITLE
Add async operations processing

### PR DESCRIPTION
**Pull request by aniezurawski**

### CODE REVIEW

File: neptune/internal/containers/disk_queue.py [hunk 1]
```diff
@@ -83,6 +84,11 @@ def get_batch(self, size: int) -> List[T]:
     def flush(self):
         self._writer.flush()
 
+    def is_overflowing(self) -> bool:
```
Comment (PiotrJander) [inline neptune/internal/containers/disk_queue.py:87]: How is this used?

File: neptune/internal/operation.py [hunk 2]
```diff
@@ -81,18 +146,52 @@ def __init__(self, exp_uuid: uuid.UUID, path: List[str], values: List[str]):
     def accept(self, visitor: 'OperationVisitor[Ret]') -> Ret:
         return visitor.visit_log_strings(self)
 
+    def to_dict(self) -> dict:
+        return {
+            "type": self.__class__.__name__,
+            "exp_uuid": str(self.exp_uuid),
+            "path": self.path,
+            "values": self.values
+        }
```
Comment (PiotrJander) [inline neptune/internal/operation.py:155]: A lot of repetition here, might be worth thinking about deduplication

File: neptune/internal/variable_setter_value_visitor.py [hunk 3]
```diff
@@ -30,38 +30,41 @@
...
         return var
 
     def visit_string_series(self, value: StringSeries) -> Variable:
         var = StringSeriesVariable(self._experiment, self._path)
         var.clear()
         # TODO: Avoid loop
-        for val in value.values:
+        for val in value.values[:-1]:
             var.log(val)
+        var.log(value.values[len(value.values) - 1], self._wait)
```
Comment (PiotrJander) [inline neptune/internal/variable_setter_value_visitor.py:63]: value.values[-1] ?

**Review by PiotrJander — APPROVED** [review]

File: neptune/internal/containers/disk_queue.py (hunk 1 above)
Comment (aniezurawski) [reply neptune/internal/containers/disk_queue.py:87]: Consumer is woken up before its usual sleep time has expired if there is too many operations in the queue.

File: neptune/internal/operation_processor.py [hunk 4]
```diff
@@ -13,23 +13,17 @@
...
-class Queue(Generic[T]):
 
-    @abc.abstractmethod
-    def put(self, obj: T) -> None:
-        pass
+class OperationProcessor:
 
     @abc.abstractmethod
-    def get(self) -> T:
+    def queue_operation(self, op: Operation, wait: bool) -> None:
```
[the remark this answers is not in the document]
Comment (aniezurawski) [reply neptune/internal/operation_processor.py:24]: Ok.

File: neptune/internal/containers/queue.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+# limitations under the License.
+#
+
+import abc
+from typing import Generic, TypeVar, List
+
+T = TypeVar('T')
+
+
+class Queue(Generic[T]):
```
[the remark this answers is not in the document]
Comment (aniezurawski) [reply neptune/internal/containers/queue.py:23]: Ok. Any suggestions?

File: neptune/internal/variable_setter_value_visitor.py (hunk 3 above)
Comment (aniezurawski) [reply neptune/internal/variable_setter_value_visitor.py:63]: Ok.

File: neptune/internal/containers/queue.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (PiotrJander) [reply neptune/internal/containers/queue.py]: NeptuneQueue, DiskQueue? Not feeling inspired

**Pull request merged**